pandas/io/json/_normalize.py

```diff
@@ -153,7 +153,7 @@ def _normalise_json(
                 # to avoid adding the separator to the start of every key
                 # GH#43831 avoid adding key if key_string blank
                 key_string=new_key
-                if new_key[: len(separator)] != separator
+                if key_string or new_key[: len(separator)] != separator
```

Comment:
Does `key_string` being false imply that

```
new_key[: len(separator)] == separator
```
?

If so, then can this be simplified to just `if key_string`?

Reply:
The implication if set is that you are within at least one recursive call. Seems like the string substitution in place should only effect the very top of the hierarchy. 

Probably a cleaner way to represent it - this was just a quick bolt on to the existing code

Reply:
Sorry not sure I understand the reply  (or perhaps my suggestion was unclear) - I was suggesting:
```python
            _normalise_json(
                data=value,
                # to avoid adding the separator to the start of every key
                # GH#43831 avoid adding key if key_string blank
                key_string=new_key if key_string else removeprefix(new_key, separator),
                normalized_dict=normalized_dict,
                separator=separator,
            )
```
because if `key_string` is falsey, then `new_key[: len(separator)] != separator` must also be falsey, and so the latter isn't needed (if you have `a or b` and you know `not a implies not b`, then `a or b` is the same as `a`). Wouldn't this also be a quick bolt-on to the existing code, but simpler?

Reply:
Nice idea. We still support Python 3.8 though right? I think removeprefix was added in 3.9

Can also move this out of the argument list if that helps readability - even the way it was I agree is less than desirable with readability

Reply:
True, but there's a 3.8 version of it in pandas, which is used in some places, e.g.

https://github.com/pandas-dev/pandas/blob/4a5d77f03bad3502a6504b03922807165881c630/pandas/core/ops/common.py#L56-L64

If you write it like that (with the `if sys.version_info < (3, 9):` check) then `pyupgrade` will automatically only keep the 3.9+ version when pandas drops 3.8

Reply:
Ah very nice. Cool let me do a refactor with this should make things cleaner